[python] Package "click" has a new version which breaks linting in CI

### 👓 What did you see?

python CI broke on using pyflakes, I think, and in doing so, reported a traceback.

> ImportError: cannot import name '_unicodefun' from 'click'

This issue in another Python-using project describes the problem:
https://github.com/dask/distributed/issues/6013

```
black....................................................................Failed
- hook id: black
- exit code: 1

Traceback (most recent call last):
  File "/home/runner/.cache/pre-commit/repoa_4yao8n/py_env-python3.10/bin/black", line 8, in <module>
    sys.exit(patched_main())
  File "/home/runner/.cache/pre-commit/repoa_4yao8n/py_env-python3.10/lib/python3.10/site-packages/black/__init__.py", line 1372, in patched_main
    patch_click()
  File "/home/runner/.cache/pre-commit/repoa_4yao8n/py_env-python3.10/lib/python3.10/site-packages/black/__init__.py", line 1358, in patch_click
    from click import _unicodefun
ImportError: cannot import name '_unicodefun' from 'click' (/home/runner/.cache/pre-commit/repoa_4yao8n/py_env-python3.10/lib/python3.10/site-packages/click/__init__.py)

flake8...................................................................Passed
Error: Process completed with exit code 1.
```

### ✅ What did you expect to see?

A passing CI for Python.

### 📦 Which tool/library version are you using?

Well, this is here, in this repo, in CI.

### 🔬 How could we reproduce it?

See 
https://github.com/cucumber/cucumber-expressions/runs/5733063101?check_suite_focus=true

### 📚 Any additional context?

--- a/python/.pre-commit-config.yaml
+++ b/python/.pre-commit-config.yaml
@@ -1,6 +1,6 @@
 repos:
 -   repo: https://github.com/psf/black
-    rev: 21.12b0
+    rev: 22.3.0
     hooks:
     -   id: black
 -   repo: https://github.com/PyCQA/flake8
@@ -8,4 +8,3 @@ repos:
     hooks:
     -   id: flake8
         args: ['--max-line-length=130','--ignore=E203,W503']
-

--- a/python/tests/test_regular_expression.py
+++ b/python/tests/test_regular_expression.py
@@ -57,13 +57,10 @@ class TestRegularExpression:
         ) == ["Charlie"]
 
     def test_ignores_non_capturing_groups(self):
-        assert (
-            self._match(
-                r"(\S+) ?(can|cannot) (?:delete|cancel) the (\d+)(?:st|nd|rd|th) (attachment|slide) ?(?:upload)?",
-                "I can cancel the 1st slide upload",
-            )
-            == ["I", "can", 1, "slide"]
-        )
+        assert self._match(
+            r"(\S+) ?(can|cannot) (?:delete|cancel) the (\d+)(?:st|nd|rd|th) (attachment|slide) ?(?:upload)?",
+            "I can cancel the 1st slide upload",
+        ) == ["I", "can", 1, "slide"]
 
     def test_matches_capture_group_nested_in_optional_one(self):
         regexp = r"^a (pre-commercial transaction |pre buyer fee model )?purchase(?: for \$(\d+))?$"

